utils/__tests__/profile.test.ts

```diff
@@ -258,14 +285,109 @@ describe("getLimitedProfileTask", () => {
+    ${"the SPECIAL service has INACTIVE activation"}                     | ${"not allow"} | ${"MANUAL"} | ${some(aRetrievedProfileWithManualPreferences)} | ${some({ ...anActivation, status: ActivationStatusEnum.INACTIVE })}                                              | ${false}
+    ${"the SPECIAL service has INACTIVE activation"}                     | ${"not allow"} | ${"AUTO"}   | ${some(aRetrievedProfileWithAutoPreferences)}   | ${some({ ...anActivation, status: ActivationStatusEnum.INACTIVE })}                                              | ${false}
+    ${"the SPECIAL service has PENDING activation in grace period"}      | ${"allow"}     | ${"AUTO"}   | ${some(aRetrievedProfileWithAutoPreferences)}   | ${some({ ...anActivation, status: ActivationStatusEnum.PENDING, _ts: Date.now() })}                              | ${true}
+    ${"the SPECIAL service has PENDING activation outside grace period"} | ${"not allow"} | ${"AUTO"}   | ${some(aRetrievedProfileWithAutoPreferences)}   | ${some({ ...anActivation, status: ActivationStatusEnum.PENDING, _ts: subSeconds(Date.now(), mockGracePeriod) })} | ${false}
+    ${"the SPECIAL service has not an activation"}                       | ${"not allow"} | ${"MANUAL"} | ${some(aRetrievedProfileWithManualPreferences)} | ${none}                                                                                                          | ${false}
+    ${"the SPECIAL service has not an activation"}                       | ${"not allow"} | ${"AUTO"}   | ${some(aRetrievedProfileWithAutoPreferences)}   | ${none}                                                                                                          | ${false}
+  `(
+    "should $allowOrNot a sender if the user uses $mode subscription mode and $preferencesConfiguration",
+    async ({ maybeProfile, maybeActivation, expected }) => {
+      mockProfileFindLast.mockImplementationOnce(() => TE.of(maybeProfile));
+      mockServiceActivationFind.mockImplementationOnce(() =>
+        TE.of(maybeActivation)
+      );
+
+      const result = await getLimitedProfileTask(
+        anAzureApiAuthorization,
+        {
+          ...anAzureUserAttributes,
+          service: {
+            ...anAzureUserAttributes.service,
+            serviceMetadata: {
+              ...anAzureUserAttributes.service.serviceMetadata,
+              scope: ServiceScopeEnum.NATIONAL,
+              category: SpecialServiceCategoryEnum.SPECIAL
+            }
+          }
+        },
+        aFiscalCode,
+        mockProfileModel,
+        false,
+        [],
+        mockServicePreferenceModel,
+        mockServiceActivationModel,
+        canSendMessageOnActivation,
+        mockTelemetryClient
+      )();
+      result.apply(mockExpressResponse);
+
+      expect(mockServicePreferenceModel.find).not.toHaveBeenCalled();
```

Comment:
I'd check that `mockServiceActivationModel` has been called with something. In normal services tests we checked that it was not called, for special services maybe we can check that it is actually called.

Reply:
fixed in [dec75db](https://github.com/pagopa/io-functions-services/pull/215/commits/dec75dbd5fcfb899e0321cd85c761edf3982c68a)